test/parallel/test-event-on-async-iterator.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const { on, EventEmitter } = require('events');
+
+async function basic() {
+  const ee = new EventEmitter();
+  process.nextTick(() => {
+    ee.emit('foo', 'bar');
+    ee.emit('foo', 42);
```

Comment:
Emit another event and make sure we don't get notified about it?

Reply:
how can we make sure we don't get notified?

Reply:
Sorry I was unclear - by "we don't get notified" I mean "Iterating the AsyncIterator does not return it as one of the values"

Reply:
How would I test that? Maybe I'm just too tired.

Reply:
You'd do `ee.emit('bar', 15);` between the two `foo` emits and the rest of the test would stay exactly the same